test/es-module/test-esm-detect-ambiguous.mjs

```diff
@@ -306,35 +326,48 @@ describe('Module syntax detection', { concurrency: !process.env.TEST_PARALLEL },
         fixtures.path('es-modules/package-without-type/commonjs-wrapper-variables.js'),
       ]);
 
-      strictEqual(stderr, '');
-      strictEqual(stdout, 'exports require module __filename __dirname\n');
+      if (stderr) {
```

Comment:
Why would there be an `if` for an assertion? This is a test: it should always have one exact expected result, not two possible results that we need a conditional for.